### PR TITLE
Add links and meta to relationships and resources

### PR DESCRIPTION
**Pull request by TheWudu**
Add the links to relationships if the include option is passed.

### CODE REVIEW

File: lib/active_model/serializer/adapter/json_api.rb [hunk 1]
```diff
@@ -174,6 +174,11 @@ def add_resource_relationships(attrs, serializer, options = {})
               end
             end
 
+            if association.respond_to?(:json_api_links) &&
+               @options[:include].include?(name.to_s)
+              add_links(attrs[:relationships][name], association)
```
Comment (groyoh) [inline lib/active_model/serializer/adapter/json_api.rb]: I don't think this is correct. Links on a relationships might be different than links on a resource. For example, on a relationship we might have `"self": "/articles/1/relationships/tags"` and on the resource we might have `"self": "/tags/1"`.

Comment (TheWudu) [inline lib/active_model/serializer/adapter/json_api.rb]: That's right, so we have to differentiate between "relationship_links" and "resource_links" ?! 

then i would rename "json_api_links" to "resource_links"
and use the new one: "relationship_links" for this ?

Comment (groyoh) [inline lib/active_model/serializer/adapter/json_api.rb]: Yes, I think that's the way to go.

**Comment by groyoh** [comment]
Looks like there are specs failling, could you have a look?

**Comment by groyoh** [comment]
Could you also add test for these new features?

**Pull request merged**